packages/module/src/FieldBuilder/FieldBuilder.tsx

```diff
@@ -0,0 +1,330 @@
+  rowCount: number;
+  /**
+   * A function that returns the content for each row. This "render prop" provides
+   * maximum flexibility for defining the inputs within each row.
+   * Can return 1 child (single-column) or 2 children (two-column).
+   */
+  children: (helpers: FieldRowHelpers) => React.ReactNode;
+  /** A callback triggered when the "Add" button is clicked. */
+  onAddRow: () => void;
+  /** A callback triggered when a "Remove" button is clicked, which receives the index of the row to remove. */
+  onRemoveRow: (index: number) => void;
+  /** Optional props to customize the "Add" button. */
+  addButtonProps?: Omit<ButtonProps, 'onClick'>;
+  /**
+   * Optional label prefix for each row group. Defaults to "Row".
+   * Screen readers will announce this as "Row 1", "Row 2", etc.
+   */
+  rowGroupLabelPrefix?: string;
+  /**
+   * Optional unique ID prefix for this FieldBuilder instance.
+   * This ensures unique IDs when multiple FieldBuilders exist on the same page.
+   */
+  fieldBuilderIdPrefix?: string;
+}
+
+/**
+ * FieldBuilder is a component group that simplifies the creation of dynamic,
+ * multi-row forms with a consistent layout. It manages the layout and actions
+ * for adding and removing rows, while giving the consumer full control over the fields themselves.
+ */
+export const FieldBuilder: FunctionComponent<FieldBuilderProps> = ({
+  firstColumnLabel,
+  secondColumnLabel,
+  rowCount,
+  children,
+  onAddRow,
+  onRemoveRow,
+  addButtonProps = {},
+  rowGroupLabelPrefix = "Row",
+  fieldBuilderIdPrefix = "field-builder",
+  ...formGroupProps
+}: FieldBuilderProps) => {
+  // Track the previous row count to detect when rows are added
+  const prevRowCountRef = useRef(rowCount);
+  // Track focusable elements for each row
+  const focusableElementsRef = useRef<Map<number, HTMLElement>>(new Map());
+  // Track remove button refs for focus management
+  const removeButtonRefs = useRef<Map<number, HTMLButtonElement>>(new Map());
+  // Track the add button ref
+  const addButtonRef = useRef<HTMLButtonElement>(null);
+  // ARIA live region for announcing dynamic changes
+  const liveRegionRef = useRef<HTMLDivElement>(null);
+
+  // Generate unique IDs for this instance
+  const instanceId = useRef(`${fieldBuilderIdPrefix}-${Math.random().toString(36).substr(2, 9)}`);
+  const firstColumnLabelId = `${instanceId.current}-first-column-label`;
+  const secondColumnLabelId = secondColumnLabel ? `${instanceId.current}-second-column-label` : undefined;
+
+  // Function to announce changes to screen readers
+  const announceChange = useCallback((message: string) => {
+    if (liveRegionRef.current) {
+      liveRegionRef.current.textContent = message;
+      // Clear the message after a delay to prepare for next announcement
+      setTimeout(() => {
+        if (liveRegionRef.current) {
+          liveRegionRef.current.textContent = '';
+        }
+      }, 1000);
+    }
+  }, []);
```

Comment:
Rather than having to create a ref and do this, we could just capture some state that gets rendered as children to the live region div. Then in here we update that state to the message or an empty string.